implementations/jmc/generate-and-run.sh

```diff
@@ -0,0 +1,134 @@
+#! /bin/bash
+
+# script parameters
+if [ $# -eq 1 ] ; then
+    dir=$1
+    SCHEMA="$dir/schema-noformat.json"
+    INSTANCES="$dir/instances.jsonl"
+    BACKEND=C
+    LOOP=100
+elif [ $# -eq 2 ] ; then
+    SCHEMA=$1 INSTANCES=$2 BACKEND=C LOOP=100
+elif [ $# -eq 3 ] ; then
+    SCHEMA=$1 INSTANCES=$2 BACKEND=$3 LOOP=100
+elif [ $# -eq 4 ] ; then
+    SCHEMA=$1 INSTANCES=$2 BACKEND=$3 LOOP=$4
+else
+    echo "unexpected parameters" >&2
+    exit 2
+fi
+
+# backend configuration
+case $BACKEND in
+    C|c)
+        backend=c
+        bench=./model.out
+        ;;
+    PY|Py|Python|py|python)
+        backend=python
+        bench=./model.py
+        ;;
+    JS|js|javascript|JavaScript)
+        backend=javascript
+        bench=./model.js
+        ;;
+    *)
+        echo "unexpected backend: $BACKEND" >&2
+        exit 3
+        ;;
+esac
+
+NAME=$(basename $(dirname $SCHEMA))
+
+function H
+{
+    echo "# $NAME" "$@" >&2
+}
+
+H schema: $SCHEMA
+H instances: $INSTANCES
+H backend: $backend
+H loop: $LOOP
+
+#
+# SPECIAL CASE HANDLING
+#
+case $NAME in
+    cspell|ui5-manifest)
+        H uselessly re2 incompatible regex
+        if [ "$backend" = "c" ] ; then
+            jmc_opt="-re pcre2"
+        else
+            jmc_opt="-re re"
+        fi
+        jsu_model_opt=
+        ;;
+    openapi)
+        jmc_opt=
+        jsu_model_opt="--no-id"
+        ;;
+    *)
+        jmc_opt=
+        jsu_model_opt=
+        ;;
+esac
+
+source /venv/bin/activate
+rm -f model.json $bench
+
+#
+# COMPILE
+#
+H schema size: $(stat -c '%s' "$SCHEMA")
+
+let compile_start=$(date +%s.%N | tr -d .)
+
+# generate model from schema, by id or strict conversion
+H generating...
+jsu-simpler "$SCHEMA" | jsu-model --id --loose $jsu_model_opt > model.json
+status=$?
+
+if [ $status -eq 0 ] ; then
+    # generate exec from model
+    H compiling...
+    jmc --loose-number -D JSONSCHEMA_BENCHMARK \
+        --maps "https://json-model.org/models/ /app/json-model/models/" \
+        $jmc_opt -o $bench model.json
+    status=$?
+fi
+
+let compile_end=$(date +%s.%N | tr -d .)
+let compile_time=$(( $compile_end - $compile_start))
+
+H compile time: $(( $compile_time / 1000 )) µs
+[ $status -ne 0 ] && exit $status
+
+#
+# BENCH
+#
+H benchmarking...
+H instances size: $(cat "$INSTANCES" | wc -lc)
+
+# one direct run to collect pass/fail
+let run_start=$(date +%s.%N | tr -d .)
+$bench --jsonl "$INSTANCES" > model.txt
+status=$?
+let run_end=$(date +%s.%N | tr -d .)
+let run_time=$(( $run_end - $run_start))
+
+pass=$(grep PASS model.txt | wc -l)
+fail=$(grep FAIL model.txt | wc -l)
+err=$(grep ERROR model.txt | wc -l)
+
+H results: pass=$pass fail=$fail error=$err
```

Comment:
Note that if there are any failures, the process should exit with a non-zero status code.

Reply:
It should, because the commands status recovered on line 115 is non zero if not eveything passed, and is used to exit on line 134.

Anyway, I added  a consistency check with the output when the first command claims success, and another status check on the actual benchmarking run.